test/structured_metadata_spec.js

```diff
@@ -0,0 +1,160 @@
+        metadata_field_external_ids.push(result.external_id);
+      });
+    });
+    it("should not create date field with illegal default value", () => {
+      metadata.default_value = invalidDate;
+      return api.create_metadata_field(metadata).then(() => {
+        expect().fail();
+      }).catch((res) => {
+        expect(res.error).not.to.be(void 0);
+        expect(res.error.message).to.contain("default_value is invalid");
+      });
+    });
+  });
+  it("should return list metadata field definitions", function () {
+    return api.list_metadata_fields().then((result) => {
+      expect(result).not.to.be.empty();
+      expect(result.metadata_fields).not.to.be.empty();
+      result.metadata_fields.forEach((field) => {
+        expect(field).to.include.keys(...mandatory_fields);
+      });
+    });
+  });
+  it("should return metadata field by external id", function () {
+    return api.get_metadata_field(metadata_arr[0].external_id).then((result) => {
+      expect(result).not.to.be.empty();
+      expect(result).to.include.keys(...mandatory_fields);
+      expect(result.label).to.eql(metadata_arr[0].label);
+    });
+  });
+  it("should update metadata field by external id", function () {
+    const metadata = {
+      default_value: "red",
+    };
+    return api.update_metadata_field(metadata_arr[0].external_id, metadata).then((result) => {
+      expect(result).not.to.be.empty();
+      expect(result).to.include.keys(...mandatory_fields);
+      expect(result.label).to.eql(metadata_arr[0].label);
+      expect(result.default_value).to.eql(metadata.default_value);
+    });
+  });
+  it("should delete metadata field by external id", function () {
+    return api.delete_metadata_field(metadata_arr[0].external_id).then((result) => {
+      expect(result).not.to.be.empty();
+      expect(result.message).to.eql("ok");
+    });
+  });
+  it("should update metadata field datasource by external id", function () {
+    const datasource = {
+      values: [
+        { external_id: "color_1", value: "brown" },
+        { external_id: "color_2", value: "black" },
+      ],
+    };
+    return api.update_metadata_field_datasource(metadata_arr[1].external_id, datasource).then((result) => {
+      expect(result).not.to.be.empty();
+      expect(result.values).not.to.be.empty();
+      result.values.forEach((item) => {
+        let before_update_value = datasource.values.find(val => val.external_id === item.external_id);
+        expect(item.value).to.eql(before_update_value.value);
+      });
+    });
+  });
+  it("should delete entries in metadata field datasource", function () {
+    const external_ids = [metadata_arr[1].datasource.values[0].external_id];
+    return api.delete_entries_field_datasource(metadata_arr[1].external_id, { external_ids }).then((result) => {
+      expect(result).not.to.be.empty();
+    });
+  });
+  it.skip("should upload image with metadata option", function () {
```

Comment:
Why is this test marked as skip? Also, the URL used for this test should not rely on a temporary cloud

Reply:
This test failed and was thus marked `skip` because the metadata feature flag wasn't activated for our cloud.

It has since been activated and uploading works.
We also fixed what you asked for here.